skimage/feature/tests/test_corner.py

```diff
@@ -229,6 +230,34 @@ def test_hessian_matrix_eigvals_3d(im3d, dtype):
     assert np.max(response0) > 0
 
 
+def _reference_eigvals_computation(S_elems):
+    """Legacy eigenvalue implementation based on np.linalg.eigvalsh.
+
+    Used for testing 2D and 3D analytical cython eigenvalue computations.
+    """
+    matrices = _symmetric_image(S_elems)
+    # eigvalsh returns eigenvalues in increasing order. We want decreasing
+    eigs = np.linalg.eigvalsh(matrices)[..., ::-1]
+    leading_axes = tuple(range(eigs.ndim - 1))
+    eigs = np.transpose(eigs, (eigs.ndim - 1,) + leading_axes)
+    return eigs
+
+
+@pytest.mark.parametrize(
+    'shape', [(64, 64), (512, 1024), (8, 16, 24)]
+)
+@pytest.mark.parametrize('dtype', [np.float32, np.float64])
+def test_hessian_eigvals_analytical(shape, dtype):
+    rng = np.random.default_rng(seed=5)
+    img = rng.integers(0, 256, shape)
+    H = hessian_matrix(img)
+    H = tuple(h.astype(dtype, copy=False) for h in H)
+    evs1 = _reference_eigvals_computation(H)
+    evs2 = hessian_matrix_eigvals(H)
+    atol = 1e-10
```

Comment:
Do you know if the new algorithm is more or less stable numerically?

Reply:
Unfortunately, I don't know for sure. In the 2D case, it is the same analytical forumula we were already using previously, so I think that one should be equivalent. For 3D, I assume there is some difference due to use of a general solver previously vs. the analytical formula here, but don't know how best to evaluate that.